Cryptogravisor.java

```diff
@@ -1,43 +1,77 @@
+import javax.swing.*;
+import java.util.ArrayList;
+import java.math.BigInteger;
+import java.security.SecureRandom;
 public class Cryptogravisor extends JFrame {
     //the main class of the program
     //handles all the GUI, and calls methods in other classes
     
-    private ArrayList<Encryption> encryptlist;
-    private Contacts;
-    private Cryptogravisor() {
+    private ArrayList<Encryption> encryptlist = new ArrayList<Encryption>();
+    private Contacts contacts = new Contacts();
+    private Communications comm;
+    public Cryptogravisor() {
+	super ("ayylmao");
 	//initialize GUI and stuff
+	comm=new Communications(6000,6001,this);
+	setVisible(true);
     }
     
-    private void actionPerformed() {
+    public void actionPerformed() {
 	//responds to user input
 	//delegates message sends, and user adds
 
     }
     private void addToContacts(String name, String address) {
-	//adds the user entry to the contacts class
+    	// Creates the big integer
+    	BigInteger bigint = Encryption.calcMod();
+    	// adds the user entry to the contacts class
+    	contacts.addContact(name, address);
+    	encryptlist.add(new Encryption(bigint));
+    	// sends the key and mod to given address 
+    	comm.sendKeyAndMod(address, encryptlist.get(encryptlist.indexOf(address)).DiffieHellmanComputeKey(), bigint);
     }
     private void updateContactList() {
 	//updates GUI, to the current contents of Contacts class
     }
-    private void sendMSG() {
+    private void sendMSG(String address, String message) throws InvalidatedEncryptionException {
 	//sends messaged based on the user's selected contact and their message in the textfield
 	//encrypts for recipient, and sends to user with communications class
+    	// sends the encrypted message to the address
+	    comm.send(address, encryptlist.get(encryptlist.indexOf(address)).encryptText(message));
```

Comment:
This is not the way to get the index. You should be using the contacts class to find the index